flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/RankITCase.scala

```diff
@@ -656,18 +656,14 @@ class RankITCase(mode: StateBackendMode) extends StreamingWithStateTestBase(mode
       "(true,1,book,b,3,2)",
       "(true,2,fruit,a,2,1)",
       "(true,3,book,a,1,2)",
-      "(true,3,book,a,1,2)",
```

Comment:
Is this change expected?

Reply:
It's not expected, as the upsert key is wrong in the last commit, should be partition by key and row number instead of partition by key and order by key.  The states are stored by upsert keys, therefore wrong upsert keys will cause wrong result.